### ISSUE TITLE
parseISO throws Cannot read property 'match' of null

### ISSUE THREAD

**Issue by onoranzefunebricloud** [date-fns/date-fns#1173]
```
> parseISO("2019-05-16+02:00")

TypeError: Cannot read property 'match' of null
    at parseYear (~/node_modules/date-fns/parseISO/index.js:180:29)
    at parseISO (~/node_modules/date-fns/parseISO/index.js:106:25)
```

**Comment by ekschro** [comment]
I just tried passing your argument `"2019-05-16+02:00"` through the v1 API equivalent of `parseISO`, `parse`.  I think your format maybe invalid.

The results were as follows:
```
> const d = require('date-fns');
undefined
> d.parse('2019-05-16+02:00')
Invalid Date
```

I believe you forgot to specify the time of day.  By changing you argument to `"2019-05-16T00:00+02:00"`, it translates to **midnight of February 5th, 2016 with a time offset of +02:00**.

 When passing my changes in into `parse` I got `2019-05-15T22:00:00.000Z`.
```
> const d = require('date-fns')
> d.parse('2019-05-16T00:00+02:00')
2019-05-15T22:00:00.000Z
```

**v2** doesn't seem to be handling the error correctly and outputting `Invalid Date`.
